src/Swarrot/Processor/XDeath/XDeathMaxCountProcessor.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Swarrot\Processor\XDeath;
+
+use PhpAmqpLib\Wire\AMQPArray;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Psr\Log\NullLogger;
+use Swarrot\Broker\Message;
+use Swarrot\Broker\MessageProvider\MessageProviderInterface;
+use Swarrot\Processor\ProcessorInterface;
+use Swarrot\Processor\ConfigurableInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class XDeathMaxCountProcessor implements ConfigurableInterface
+{
+    /**
+     * @var ProcessorInterface
+     */
+    protected $processor;
+
+    /**
+     * @var MessageProviderInterface
+     */
+    private $messageProvider;
+
+    /**
+     * @var \Closure
+     */
+    private $callback;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param ProcessorInterface $processor
+     * @param MessageProviderInterface $messageProvider
+     * @param \Closure $callback
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(
+        ProcessorInterface $processor,
+        MessageProviderInterface $messageProvider,
+        \Closure $callback,
```

Comment:
should be `callable` rather than `\Closure`. I don't see any benefit restricting the kind of callable you can use (btw, when using the Symfony DI container, forcing to use closures would make it harder to configure, because you cannot directly put a closure in the container)